python/tvm/autotvm/task/space.py

```diff
@@ -838,22 +995,26 @@ def _add_new_transform(self, space_class, name, axes, policy, **kwargs):
             return [Axis(space, i) for i in range(space.num_output)]
         return [Axis(None, i) for i in range(space_class.get_num_output(axes, policy, **kwargs))]
 
-    def __len__(self):
```

Comment:
there are pros and cons to remove length. And it seems the number of modified scripts show that len(task.space) became quite public api :( it will be quite painful for endusers if we remove this function

let's return filtered_length by it and add clear comments to declaration of filtered_length and total_length when we need to use each

Reply:
Agreed, restored the original one.